protoflo/components/__init__.py

```diff
@@ -255,31 +264,28 @@ def getCache ():
 	results = []
 
 	for moduleObj in getSearchDirectories():
-
 		componentPath = moduleObj.filePath
 		dropinPath = componentPath.parent().child('components.cache')
 
-		try:
-			lastModified = componentPath.getModificationTime()
-		except:
-			log.err("Could not stat {:s}".format(str(componentPath)))
-			continue
-
 		# Look for cache
 		try:
 			lastCached = dropinPath.getModificationTime()
 			collection = pickle.load(dropinPath.open('r'))
+		# FIXME: what kind of error do we expect?
 		except:
-			lastCached = 0
-
-		
-		if lastCached < lastModified:
 			stale = True
 		else:
 			stale = False
-			for component in collection.components:
-				if FilePath(component.fileName).getModificationTime() > lastCached:
-					stale = True
+			for path in componentPath.parent().walk():
```

Comment:
Another option is to get rid of caching.  I'm not convinced that it saves that much time and I'm afraid it will be a source of future bugs.